src/twisted/protocols/test/test_tls.py

```diff
@@ -977,6 +981,54 @@ def disconnected(ign):
         return disconnectDeferred
 
 
+    def test_noCircularReferences(self):
+        """
+        TLSMemoryBIOProtocol doesn't leave circular references that make
+        it alive after connection is closed.
+        """
+        def nObjectsOfType(type):
+            return sum(1 for x in gc.get_objects() if isinstance(x, type))
+
+        gc.disable()
+
+        try:
+            class DummyServerProtocol(Protocol):
+                def dataReceived(self, data):
+                    self.transport.loseConnection()
+
+
+            class DummyClientProtocol(Protocol):
+                def connectionMade(self):
+                    self.transport.write(b'hello')
+
+                def connectionLost(self, reason):
+                    reason.trap(ConnectionDone)
```

Comment:
Just asking.  I have not executed (yet) the tests. Can the connection be closed for other reasons?
Please document why ConnectionDone is important here.